app/uk/gov/hmrc/apiplatformjobs/scheduled/DeleteUnusedApplicationsJob.scala

```diff
@@ -33,23 +34,42 @@ abstract class DeleteUnusedApplicationsJob(thirdPartyApplicationConnector: Third
                                            mongo: ReactiveMongoComponent)
   extends UnusedApplicationsJob("DeleteUnusedApplicationsJob", environment, configuration, mongo) {
 
+  def batchFutures[I](fn: I => Future[Unit])(input: Seq[I])(implicit ec: ExecutionContext): Future[Unit] = {
+    input.splitAt(20) match {
```

Comment:
Is this the batch size of 20?
I think that’s too high. If the requests were quick (50ms?), it could still be trying hundreds per second which could trip the Akamai DDos rules.

Batch size 1?

Reply:
20 is the batch size, basically limiting it to 20 concurrent operations until they all return.